dobson/utils.py

```diff
@@ -62,3 +68,33 @@ def get_mac_addresses():
 
         mac_addr = device[-18:].strip(' ').lower().replace(' ', ':')
         yield mac_addr
+
+
+def get_unknown_mac_addresses():
+    """
+    Yields MAC Addresses connected to the router which are _not_ in devices.json. Useful for
+    tagging new devices
+    """
+    for mac_addr in get_mac_addresses():
+        if mac_addr not in KNOWN_DEVICES:
+            yield mac_addr
+
+
+def add_device(mac_addr, data):
+    """Adds an entry for the specified Mac Address to devices.json and KNOWN_DEVICES
+    If the specified Mac Address already exists, returns False and does not update"""
```

Comment:
I'd format this like the others (`Mac` -> `MAC` and the triple quotes on their own lines